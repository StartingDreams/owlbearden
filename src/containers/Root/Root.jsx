import React from 'react';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
  Route,
  // Link
} from 'react-router-dom';

// import { syncHistoryWithStore } from 'react-router-redux';
import Home from '../../modules/Dashboard/Home';
import Spells from '../../modules/Casting/containers/Spells';
import createStore from '../../state';

const store = createStore();

export default function Root() {
  return (
    <Provider store={store}>
      <div id="react-root">
        <Router>
          <div>
            <Route exact path="/" component={Home} />
            <Route path="/casting/spells" component={Spells} />
          </div>
        </Router>
      </div>
    </Provider>
  );
}
