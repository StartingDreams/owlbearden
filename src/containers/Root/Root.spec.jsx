import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Root from './Root';

jest.mock('../../state', () => ({
  createStore: jest.fn(),
}));

jest.mock('react-redux', () => ({
  Provider: 'Provider',
}));

jest.mock('react-router-dom', () => ({
  BrowserRouter: 'BrowserRouter',
  Route: 'Route',
}));

jest.mock('../../styleguide', () => ({
  Theme: 'Theme',
}));

jest.mock('material-ui/styles/MuiThemeProvider', () => (function MuiThemeProvider() {}));

jest.mock('material-ui/Paper', () => (function Paper() {}));

jest.mock('../../modules/OwlBearBar', () => (function OwlBearBar() {}));

jest.mock('../../modules/Dashboard', () => (function Dashboard() {}));

jest.mock('../../components/NoMatch', () => (function NoMatch() {}));

jest.mock('../../modules/Firebase', () => (function Firebase() {}));

jest.mock('../../modules/Firebase/containers/Login', () => (function Login() {}));

jest.mock('../../modules/Casting/containers/Spells', () => (function Spells() {}));

jest.mock('../../components/PrivateRoute', () => (function PrivateRoute() {}));

describe('<Root />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Root />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
