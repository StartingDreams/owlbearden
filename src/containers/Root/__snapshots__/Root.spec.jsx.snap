// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should match the snapshot 1`] = `
<Provider>
  <div
    id="react-root"
  >
    <MuiThemeProvider
      muiTheme="Theme"
    >
      <div>
        <BrowserRouter>
          <div>
            <Firebase />
            <Header />
            <Paper
              style={
                Object {
                  "margin": 0,
                  "padding": "10px",
                }
              }
              zDepth={1}
            >
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <PrivateRoute
                component={[Function]}
                exact={true}
                path="/dashboard"
              />
              <PrivateRoute
                component={[Function]}
                exact={true}
                path="/casting/spells"
              />
              <Route
                component={[Function]}
              />
            </Paper>
            <Sidebar />
          </div>
        </BrowserRouter>
      </div>
    </MuiThemeProvider>
  </div>
</Provider>
`;
